src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -1016,6 +1016,12 @@ private static bool IsPackageReferenceAutoReferenced(PackageReference package)
                         preferredVersions.Remove(packageId);
                     }
                 }
+                else
+                {
+                    // we return an empty list here because this is trying to update a specific package in a project here which is not even installed in the project.
```

Comment:
Can we reword this? :)
I'm not sure what this is supposed to mean

Reply:
Ok, upon 2nd read and a look at the code, i got what you're trying to do (still think it can be reworded better). 

Why do we even need to call into this method at all when a package identity has been specified? 

Can it be done on a higher level?